cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java

```diff
@@ -136,6 +139,107 @@ public void testAllTypesEvaluationSuccess() throws Exception {
         }
     }
 
+    private CqlEvaluationRequest makeEvaluationRequest(String contextName) {
+        return makeEvaluationRequest(contextName, null, null);
+    }
+    
+    private CqlEvaluationRequest makeEvaluationRequest(String contextName, String libraryName, String libraryVersion) {
+        CqlEvaluationRequest cqlEvaluationRequest = new CqlEvaluationRequest();
+        cqlEvaluationRequest.setContextKey(contextName);
+        cqlEvaluationRequest.setContextValue("NA");
+
+        CqlLibraryDescriptor descriptor = new CqlLibraryDescriptor();
+        descriptor.setLibraryId(libraryName);
+        descriptor.setVersion(libraryVersion);
+        
+        cqlEvaluationRequest.setDescriptor(descriptor);
+        
+        return cqlEvaluationRequest;
+    }
+
+    @Test
+    public void testFilterRequestsNoEvaluationRequests() {
+        CqlEvaluationRequests requests = new CqlEvaluationRequests();
+
+        List<CqlEvaluationRequest> cqlEvaluationRequests = evaluator.filterRequests(requests, "A");
+
+        assertTrue(cqlEvaluationRequests.isEmpty());
+    }
+    
+    @Test
+    public void testFilterRequestsNoEvaluationRequestsForContext() {
+        CqlEvaluationRequests requests = new CqlEvaluationRequests();
+
+        List<CqlEvaluationRequest> evaluations = Arrays.asList(
+                makeEvaluationRequest("B"),
+                makeEvaluationRequest("C"),
+                makeEvaluationRequest("D")
+        );
+        
+        requests.setEvaluations(evaluations);
+        
+        List<CqlEvaluationRequest> cqlEvaluationRequests = evaluator.filterRequests(requests, "A");
+        
+        assertTrue(cqlEvaluationRequests.isEmpty());
+    }
+
+    @Test
+    public void testFilterRequestsFilterToRequestsForContext() {
+        CqlEvaluationRequests requests = new CqlEvaluationRequests();
+
+        List<CqlEvaluationRequest> evaluations = Arrays.asList(
+                makeEvaluationRequest("A"),
+                makeEvaluationRequest("A"),
+                makeEvaluationRequest("B")
+        );
+
+        requests.setEvaluations(evaluations);
+
+        List<CqlEvaluationRequest> cqlEvaluationRequests = evaluator.filterRequests(requests, "A");
+
+        assertEquals(2, cqlEvaluationRequests.size());
```

Comment:
Pedantic: Not required, but some assertion to ensure the two elements in the list are the correct ones would make this test more robust.